### PR TITLE
CA-183: feat: implement DeletionConfirmationBottomSheet for project deletion

### PR DESCRIPTION
**Pull request by ayanasamuel8**
### 1. PR SUMMARY

Implements `DeletionConfirmationBottomSheet` (DASH-031) — the confirmation sheet shown before a project is permanently deleted. The sheet warns the admin about data loss, optionally shows a count of images attached to the project's calculations, and provides "Yes, Delete" / "No, Keep" buttons. It is a pure presentational widget with callbacks, following the same structure as `DeleteEstimationConfirmationSheet`.

Key decisions:
- `imagesAttachedCount: int?` — chip is hidden when null; real count deferred to CA-269 (cost_files table not yet implemented)
- No static `show()` helper — blocked by CA-116 (ProjectDetailScreen doesn't exist yet); TODO comment left
- No new bloc events — existing `ProjectSettingsDeleteRequested` covers confirmed deletion once wired

### 2. REVIEW SUMMARY TABLE

| Rule | Issue | Status | Notes |
|---|---|---|---|
| 4 – CoreUI constants | `size: 32` in `CoreIconWidget` should use `CoreSpacing.space8` | ✅ Agreed, addressed | Changed to `CoreSpacing.space8` in commit 3 |
| 4 – CoreUI constants | `width: 60, height: 60` for icon circle — no CoreSpacing equivalent (space12=48, space16=64) | ✅ Agreed, follows pattern | Matches `DeleteEstimationConfirmationSheet` exactly; no CoreSpacing constant exists for 60 |
| 14 – Accessibility | New interactive bottom sheet lacked a11y tests for its two buttons | ✅ Agreed, addressed | Added `deletion_confirmation_bottom_sheet_a11y_test.dart`; both buttons pass tap-target + label guidelines in light and dark themes |
| Visual – Figma fidelity | Images-attached chip stretched full column width instead of hugging content | ✅ Agreed, addressed | Removed `Center` wrapper; chip now sizes to text via symmetric vertical padding (commit 2) |

## Test plan
- [x] Widget tests pass: button callbacks fire correctly, chip renders/hides based on null, edge cases (empty name, null callbacks, large count) handled
- [x] A11y tests pass: "Yes, Delete" and "No, Keep" buttons meet tap-target and label guidelines in light and dark themes
- [x] Screenshot goldens generated: isolated at 390×420 (default, with 25 images), long name at 390×500, modal-over-background at 390×844
- [x] `fvm dart run custom_lint` — no issues
- [x] `fvm flutter test` — all tests pass, no regressions

### CODE REVIEW

File: lib/l10n/app_en.arb [hunk 1]
```diff
@@ -742,6 +742,20 @@
...
+  "@deleteProjectConfirmTitle": {
+    "description": "Title for delete project confirmation bottom sheet",
+    "placeholders": {
+      "projectName": {
+        "type": "String",
+        "example": "Material of Building"
+      }
+    }
+  },
+  "deleteProjectWarningMessage": "By removing you will loose the all calculation & Cost of this project as well as image attach to your calculation",
```
Comment (ripplearcgit) [inline lib/l10n/app_en.arb]: 🔴 **User-facing copy errors that will ship to production**

Three errors in `deleteProjectWarningMessage`:
- `"loose"` → `"lose"` ("loose" means not tight; "lose" means to no longer have)
- `"the all calculation"` → `"all calculations"` (extra "the", missing plural)
- `"image attach"` → `"images attached"` (plural, past-tense participle)

Suggested replacement:
```arb
"deleteProjectWarningMessage": "By removing this project you will lose all calculations, costs, and images attached to your calculations",
```

The rewritten string is shorter, grammatically correct, and more parallel in structure. The generated `app_localizations.dart` and `app_localizations_en.dart` files will update automatically after changing the ARB.

File: lib/features/project_settings/presentation/widgets/deletion_confirmation_bottom_sheet.dart [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    super.key,
+    required this.projectName,
+    this.onConfirm,
+    this.onCancel,
+    this.imagesAttachedCount,
+  });
+
+  // TODO: CA-116 — add a static show() helper wired to ProjectSettingsBloc
+  // once ProjectDetailScreen exists and triggers this sheet.
+
```
Comment (ripplearcgit) [inline lib/features/project_settings/presentation/widgets/deletion_confirmation_bottom_sheet.dart:50]: 🔴 **TODO annotation missing mandatory brackets and YouTrack URL**

The project convention requires `[CA-XXX]` square brackets around the ticket number and a full YouTrack URL. The current format `// TODO: CA-116 —` does not satisfy the convention enforced across the codebase.

Suggested fix:
```dart
// TODO: [CA-116] Add a static show() helper wired to ProjectSettingsBloc
// once ProjectDetailScreen exists and triggers this sheet.
// https://ripplearc.youtrack.cloud/issue/CA-116
```

**Review by ripplearcgit — CHANGES_REQUESTED** [review]
### Architectural overview

```mermaid
sequenceDiagram
  participant Admin as Admin User
  participant Sheet as DeletionConfirmationBottomSheet
  participant Parent as Parent Widget (PR #384)
  Admin->>Sheet: views bottom sheet
  Note over Sheet: Shows projectName, warning text, optional image count chip
  alt onConfirm tapped (Yes, Delete)
    Sheet->>Parent: onConfirm()
    Note over Parent: Parent fires ProjectSettingsDeleteRequested
  else onCancel tapped (No, Keep)
    Sheet->>Parent: onCancel()
    Note over Parent: Parent dismisses sheet, no deletion
  else both callbacks are null
    Sheet->>Sheet: tap is gracefully no-op
  end
```

2 blocking issues require fixes before merge. Full report: ~/Desktop/pr-reviews/pr-review-construculator-app-383.html

**Review by ripplearcgit — COMMENTED** [review]
**Nits (4) — no blockers in this call:**

- `deletion_confirmation_bottom_sheet.dart` line 4 — `StatefulWidget` with no mutable state; refactor to `StatelessWidget`
- `deletion_confirmation_bottom_sheet.dart` line 108 — dead `?? 0` inside a `!= null` guard; use `widget.imagesAttachedCount!`
- `app_en.arb` line 745 — capitalised "Project" mid-sentence in `deleteProjectConfirmTitle`; use lowercase
- `deletion_confirmation_bottom_sheet_screenshot_test.dart` line 71 — long-name test duplicates `MaterialApp` setup; extract `size` param into `pumpSheet` helper

Full report: ~/Desktop/pr-reviews/pr-review-construculator-app-383.html

File: lib/features/project_settings/presentation/widgets/deletion_confirmation_bottom_sheet.dart [hunk 3]
```diff
@@ -0,0 +1,137 @@
+import 'package:construculator/libraries/extensions/extensions.dart';
+import 'package:flutter/material.dart';
+import 'package:ripplearc_coreui/ripplearc_coreui.dart';
+
```
Comment (ripplearcgit) [inline lib/features/project_settings/presentation/widgets/deletion_confirmation_bottom_sheet.dart:4]: 🟡 **Nit — unnecessary `StatefulWidget`**

The private state class `_DeletionConfirmationBottomSheetState` only overrides `build()` and never calls `setState()`. There is no local mutable state, no animation controller, and no lifecycle logic. A `StatelessWidget` is the correct base class here.

```dart
// Before
class DeletionConfirmationBottomSheet extends StatefulWidget {
  ...
  @override
  State<DeletionConfirmationBottomSheet> createState() =>
      _DeletionConfirmationBottomSheetState();
}

class _DeletionConfirmationBottomSheetState
    extends State<DeletionConfirmationBottomSheet> {
  @override
  Widget build(BuildContext context) { ... }
}

// After
class DeletionConfirmationBottomSheet extends StatelessWidget {
  final String projectName;
  final VoidCallback? onConfirm;
  final VoidCallback? onCancel;
  final int? imagesAttachedCount;

  const DeletionConfirmationBottomSheet({...});

  @override
  Widget build(BuildContext context) {
    // Access fields directly (no widget. prefix needed)
    ...
  }
}
```

File: lib/features/project_settings/presentation/widgets/deletion_confirmation_bottom_sheet.dart [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                    color: colorTheme.backgroundBlueLight,
+                    borderRadius: BorderRadius.circular(100),
+                  ),
+                  child: Text(
+                    l10n.imagesAttachedCount(widget.imagesAttachedCount ?? 0),
+                    style: textTheme.bodyMediumRegular.copyWith(
+                      color: colorTheme.textDark,
+                    ),
+                  ),
+                ),
```
Comment (ripplearcgit) [inline lib/features/project_settings/presentation/widgets/deletion_confirmation_bottom_sheet.dart:113]: 🟡 **Nit — unreachable `?? 0` null-coalescing guard**

`imagesAttachedCount ?? 0` is inside an `if (widget.imagesAttachedCount != null)` guard, so the `0` branch is dead code. Use Dart's null assertion `!` instead — it communicates intent clearly and removes the false impression that `0` can ever be passed here:

```dart
l10n.imagesAttachedCount(widget.imagesAttachedCount!)
```

File: lib/l10n/app_en.arb [hunk 5]
```diff
@@ -742,6 +742,20 @@
     "description": "Label for cancel/keep button",
     "context": "https://www.figma.com/design/vugaGpii5HfgEQHPbrS3mU/Construculator-Visual-Design?node-id=57361-9022&t=wjWHI4P6ds90viit-4"
   },
+  "deleteProjectConfirmTitle": "Are you sure you want to remove \"{projectName}\" Project?",
```
Comment (ripplearcgit) [inline lib/l10n/app_en.arb]: 🟡 **Nit — capitalised "Project" mid-sentence**

`"Are you sure you want to remove \"{projectName}\" Project?"` — "Project" is capitalised mid-sentence without justification. The rest of the UI strings in this file use sentence case. Suggest:

```arb
"deleteProjectConfirmTitle": "Are you sure you want to remove \"{projectName}\" project?",
```

File: test/features/project_settings/screenshots/deletion_confirmation_bottom_sheet_screenshot_test.dart [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+
+        await expectLater(
+          find.byType(DeletionConfirmationBottomSheet),
+          matchesGoldenFile(
+            'goldens/deletion_confirmation_bottom_sheet/${size.width.toInt()}x${size.height.toInt()}/with_images_attached.png',
+          ),
+        );
+      });
+
+      testWidgets('renders with long project name', (tester) async {
```
Comment (ripplearcgit) [inline test/features/project_settings/screenshots/deletion_confirmation_bottom_sheet_screenshot_test.dart:71]: 🟡 **Nit — long-name test duplicates `MaterialApp` setup instead of using `pumpSheet` helper**

The other two tests in the "isolated widget" group share the `pumpSheet` helper that sets `physicalSize` and `devicePixelRatio`. The long-name test rebuilds the full `MaterialApp` tree inline. Extracting a `size` parameter would remove the duplication:

```dart
Future<void> pumpSheet({
  required WidgetTester tester,
  String projectName = 'Material of Building',
  int? imagesAttachedCount,
  Size size = const Size(390, 420),  // ← make size a param
}) async {
  tester.view.physicalSize = size;
  tester.view.devicePixelRatio = ratio;
  await tester.pumpWidget(
    MaterialApp(
      theme: createTestTheme(),
      locale: const Locale('en'),
      localizationsDelegates: AppLocalizations.localizationsDelegates,
      supportedLocales: AppLocalizations.supportedLocales,
      home: Scaffold(
        body: DeletionConfirmationBottomSheet(
          projectName: projectName,
          onConfirm: () {},
          onCancel: () {},
          imagesAttachedCount: imagesAttachedCount,
        ),
      ),
    ),
  );
  await tester.pumpAndSettle();
}
```

File: lib/features/project_settings/presentation/widgets/deletion_confirmation_bottom_sheet.dart [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+              width: CoreSpacing.space10,
+              height: CoreSpacing.space1,
+              decoration: BoxDecoration(
+                color: colorTheme.textDisable,
+                borderRadius: BorderRadius.circular(2),
+              ),
+            ),
+          ),
+
+          Column(
```
Comment (ripplearcgit) [inline lib/features/project_settings/presentation/widgets/deletion_confirmation_bottom_sheet.dart]: 🔴 **CoreUI reuse: hand-rolled bottom-sheet chrome duplicates `CoreQuickSheet`**

This manually builds the exact chrome `CoreQuickSheet` (`lib/src/components/bottom_sheets/core_quick_sheet.dart` in coreui) already provides: a `Container` with `pageBackground` fill, `BorderRadius.vertical(top: Radius.circular(28))`, and a centered drag-handle bar. This is not a one-off — the same reusable primitive is composed correctly in at least four other places in this codebase: `ProjectsBottomSheet.show()`, `DateRangeBottomSheet`, `GlobalSearchPage`, and `project_creation_success_sheet.dart`, all of which wrap a *content-only* widget with `CoreQuickSheet.show(context: context, child: ...)` rather than reimplementing the decoration themselves.

This exact duplication already has a tracked ticket against the sibling widget this PR was modeled on: `cost_estimation_landing_page.dart` carries `// TODO: [CA-472] Use CoreQuickSheet.show to standardize bottom sheets`. Shipping a brand-new widget with the same anti-pattern compounds that debt instead of avoiding it — and unlike the estimation sheet, this widget has zero existing callers today, so there is no migration risk to fixing it now.

Recommended fix — since the TODO at line 24-28 already anticipates a future `show()` helper, make that helper route through `CoreQuickSheet` and strip the outer decoration/handle from `build()`, keeping only the content column:

```dart
class DeletionConfirmationBottomSheet extends StatelessWidget {
  ...
  static Future<void> show(BuildContext context, {required String projectName, ...}) {
    return CoreQuickSheet.show<void>(
      context: context,
      child: DeletionConfirmationBottomSheet(projectName: projectName, ...),
    );
  }

  @override
  Widget build(BuildContext context) {
    // No outer Container/decoration/drag-handle here — CoreQuickSheet owns that.
    return Padding(
      padding: const EdgeInsets.all(CoreSpacing.space4),
      child: Column(
        mainAxisSize: MainAxisSize.min,
        spacing: CoreSpacing.space8,
        children: [ /* icon, title, warning, chip, buttons — unchanged */ ],
      ),
    );
  }
}
```

If you disagree this is in scope for this PR, the acceptable alternative per the CoreUI-reuse convention is to reference the existing ticket explicitly — `// TODO: [CA-472]` above the `Container` — so the duplication is at least tracked rather than silently repeated.

**Review by ripplearcgit — CHANGES_REQUESTED** [review]
### Architectural overview

```mermaid
sequenceDiagram
    participant Caller as Future caller (CA-182)
    participant Sheet as DeletionConfirmationBottomSheet.build()
    participant CQS as CoreQuickSheet.show()

    rect rgb(252,232,230)
    Note over Caller,Sheet: As shipped in this PR
    Caller->>Sheet: build(context)
    Sheet->>Sheet: Container(decoration: borderRadius 28, color pageBackground)
    Sheet->>Sheet: Center(Container 40x4 drag handle)
    Sheet->>Sheet: Column(title, warning, chip, buttons)
    Note right of Sheet: Reimplements CoreQuickSheet's own decoration and handle
    end

    rect rgb(230,244,234)
    Note over Caller,CQS: Established pattern elsewhere in this repo
    Caller->>CQS: CoreQuickSheet.show(context, child: ContentOnlyWidget())
    CQS->>CQS: Container(decoration: borderRadius 28, color pageBackground)
    CQS->>CQS: Padding + drag handle
    CQS->>ContentOnlyWidget: Flexible(child)
    Note right of CQS: Used by ProjectsBottomSheet, DateRangeBottomSheet, GlobalSearchPage
    end
```

Third review pass on this PR. All 6 findings from rounds 1 and 2 are verified fixed on `c7cf4ec` with no regressions (copy fixes, TODO bracket format, StatelessWidget refactor, dead `?? 0` guard removal, lowercase copy, `pumpSheet` size param extraction). This round's independent audit surfaces one new blocking issue below — the widget's internal composition (icon/title/warning/chip/buttons) is sound and matches the sibling `DeleteEstimationConfirmationSheet`; the only structural problem is the outer chrome.

Full HTML report: ~/Desktop/pr-reviews/pr-review-construculator-app-383.html

File: test/features/project_settings/screenshots/deletion_confirmation_bottom_sheet_screenshot_test.dart [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+    group('modal over background', () {
+      const size = Size(390, 844);
+      const ratio = 1.0;
+
+      testWidgets('renders as modal over page background', (tester) async {
+        tester.view.physicalSize = size;
+        tester.view.devicePixelRatio = ratio;
+
+        await tester.pumpWidget(
+          MaterialApp(
```
Comment (ripplearcgit) [inline test/features/project_settings/screenshots/deletion_confirmation_bottom_sheet_screenshot_test.dart:99]: 🟡 **Debug banner leaks into "modal over background" golden**

This test captures `find.byType(MaterialApp)` (the whole screen) rather than a cropped widget, and the `MaterialApp` doesn't set `debugShowCheckedModeBanner: false`. The saved golden (`modal_over_background.png`) shows Flutter's red/white debug ribbon in the top-right corner — visible when the image was fetched for this review. The test's own comment says it "validates Figma overlay appearance," so the debug banner is noise that doesn't belong in a design-fidelity golden.

```dart
MaterialApp(
  debugShowCheckedModeBanner: false,
  theme: createTestTheme(),
  ...
)
```

**Review by ripplearcgit — COMMENTED** [review]
### Nits

🟡 **Missing dartdoc on public members**

Only `imagesAttachedCount` got a doc comment in round 1's fix. `projectName`, `onConfirm`, `onCancel`, and the class itself (`lib/features/project_settings/presentation/widgets/deletion_confirmation_bottom_sheet.dart`) remain undocumented. Matches the sibling `DeleteEstimationConfirmationSheet`, which has the same gap, so this isn't a regression — but the convention (all public members need `///`) applies to new files regardless of precedent. *(no single line to anchor — applies to the class declaration and its three undocumented fields)*

🟡 **Debug banner leaks into "modal over background" golden** — see inline comment on `test/features/project_settings/screenshots/deletion_confirmation_bottom_sheet_screenshot_test.dart`.

Full HTML report: ~/Desktop/pr-reviews/pr-review-construculator-app-383.html

File: lib/features/project_settings/presentation/widgets/deletion_confirmation_bottom_sheet.dart [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+                width: 60,
+                height: 60,
+                decoration: BoxDecoration(
+                  color: colorTheme.backgroundRedMid,
+                  shape: BoxShape.circle,
+                ),
+                child: Center(
+                  child: CoreIconWidget(
+                    icon: CoreIcons.delete,
+                    size: CoreSpacing.space8,
```
Comment (ripplearcgit) [inline lib/features/project_settings/presentation/widgets/deletion_confirmation_bottom_sheet.dart:80]: 🟡 **Icon size doesn't match Figma spec (non-blocking)**

Figma's `project_settings → Delete Project` node (`58447:6106`) specs the delete icon at 24×24 inside the 60×60 circle; this ships it at 32×32 (`CoreSpacing.space8`). Not a regression and not introduced by this PR — it's identical to the already-merged sibling `lib/features/estimation/presentation/widgets/delete_estimation_confirmation_sheet.dart` (which hardcodes `size: 32`), and Round 1's own reviewer asked specifically for `CoreSpacing.space8` (i.e. formalize 32 as a token, not change the value). If this is worth fixing, do it as a shared follow-up ticket touching both widgets together, not a blocker on this PR.

**Review by ripplearcgit — APPROVED** [review]
### Architectural overview

Round 2's CoreUI-reuse blocker (hand-rolled bottom-sheet chrome duplicating `CoreQuickSheet`) — resolved this round via `6fe034eb`:

```mermaid
sequenceDiagram
    participant Caller
    participant Sheet as DeletionConfirmationBottomSheet
    participant CQS as CoreQuickSheet

    rect rgb(252,232,230)
    Note over Caller,Sheet: Before this round (c7cf4ece) - flagged blocking in Round 2
    Caller->>Sheet: build(context)
    Sheet->>Sheet: Container(radius 28, pageBackground) plus Center(drag handle)
    Note right of Sheet: Reimplements CoreQuickSheet's own chrome
    end

    rect rgb(230,244,234)
    Note over Caller,CQS: This round (6fe034eb) - resolved
    Caller->>Sheet: DeletionConfirmationBottomSheet.show(context, projectName, onConfirm, onCancel, imagesAttachedCount)
    Sheet->>CQS: CoreQuickSheet.show(context, child colon this)
    CQS->>CQS: Container(radius 28, pageBackground) plus drag handle
    CQS->>Sheet: Flexible(child).build(context)
    Sheet->>Sheet: Padding(Column(icon, title, warning, chip, buttons))
    Note right of Sheet: build() now returns content only, chrome owned by CoreQuickSheet
    end
```

✅ **Verdict:** Sound fix — `build()` now returns pure content (matches CoreQuickSheet's own documented usage pattern), and the screenshot test was updated in lockstep to call the real `show()` API instead of a hand-rolled `showModalBottomSheet`.

---

**Round 3 re-review — all 10 prior findings verified fixed against HEAD, no regressions found.** See full report for the complete resolution table.

### Nits (non-blocking)
- 🟡 `deletion_confirmation_bottom_sheet.dart` line 78 — delete icon renders at 32×32 vs Figma's 24×24 spec; shared with sibling `DeleteEstimationConfirmationSheet`, optional follow-up ticket if desired, not a merge blocker. *(inline comment below)*

Full HTML report: ~/Desktop/pr-reviews/pr-review-construculator-app-383.html
